From ab3d867fe7ced1eb306af9e3ae079052c1571f0c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Pasi=20Sj=C3=B6holm?= <pasi.sjoholm@jollamobile.com>
Date: Wed, 11 Feb 2015 12:14:18 +0200
Subject: [PATCH 04/11] network: Disable IPv6 completely if method is off

If method is set off for IPv6 then it should be completely
disabled and not to allow kernel autoconfigure the addresses
for the service.
---
 src/network.c |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/network.c b/src/network.c
index 94440dc..db19cb9 100644
--- a/src/network.c
+++ b/src/network.c
@@ -650,8 +650,9 @@ static void set_connected(struct connman_network *network)
 
 	switch (ipv6_method) {
 	case CONNMAN_IPCONFIG_METHOD_UNKNOWN:
-	case CONNMAN_IPCONFIG_METHOD_OFF:
 		break;
+	case CONNMAN_IPCONFIG_METHOD_OFF:
+		__connman_ipconfig_disable_ipv6(ipconfig_ipv6);
 	case CONNMAN_IPCONFIG_METHOD_DHCP:
 	case CONNMAN_IPCONFIG_METHOD_AUTO:
 		autoconf_ipv6_set(network);
@@ -1680,8 +1681,9 @@ int __connman_network_set_ipconfig(struct connman_network *network,
 
 		switch (method) {
 		case CONNMAN_IPCONFIG_METHOD_UNKNOWN:
-		case CONNMAN_IPCONFIG_METHOD_OFF:
 			break;
+		case CONNMAN_IPCONFIG_METHOD_OFF:
+			__connman_ipconfig_disable_ipv6(ipconfig_ipv6);
 		case CONNMAN_IPCONFIG_METHOD_AUTO:
 			autoconf_ipv6_set(network);
 			break;
-- 
1.7.10.4

